packages/kbn-pm/src/commands/watch.ts

```diff
@@ -0,0 +1,110 @@
+import chalk from 'chalk';
+import { Observable, Subject } from 'rxjs';
+
+import { Project } from '../utils/project';
+import { topologicallyBatchProjects } from '../utils/projects';
+import { parallelizeBatches } from '../utils/parallelize';
+import { Command } from './';
+
+/**
+ * Name of the script in the package/project package.json file to run during `kbn watch`.
+ */
+const watchScriptName = 'kbn:watch';
+
+/**
+ * Number of milliseconds we wait before we fall back to the default watch handler.
+ */
+const defaultHandlerDelay = 3000;
+
+/**
+ * If default watch handler is used, then it's the number of milliseconds we wait for
+ * any build output before we consider watch task ready.
+ */
+const defaultHandlerReadinessTimeout = 2000;
+
+function getWatchHandlers(buildOutput$: Observable<string>) {
+  const typescriptHandler = buildOutput$
+    .first(data => data.includes('$ tsc'))
+    .map(() =>
+      buildOutput$
+        .first(data => data.includes('Compilation complete.'))
+        .mapTo('tsc')
+    );
+
+  const webpackHandler = buildOutput$
+    .first(data => data.includes('$ webpack'))
+    .map(() =>
+      buildOutput$.first(data => data.includes('Chunk Names')).mapTo('webpack')
+    );
+
+  const defaultHandler = Observable.of(void 0)
+    .delay(defaultHandlerReadinessTimeout)
+    .map(() =>
+      buildOutput$
+        .timeout(defaultHandlerDelay)
+        .catch(() => Observable.of(void 0))
+        .mapTo('timeout')
+    );
+
+  return [typescriptHandler, webpackHandler, defaultHandler];
+}
+
+function runScriptUntilWatchIsReady(project: Project) {
```

Comment:
Maybe extract this into a separate file, so it's easier to test separately from the command?

Reply:
Do you think we should have this file under `commands` or `utils` directory?

Reply:
Maybe utils to begin with? Then we can move it around if it makes sense

Reply:
ok